.github/workflows/build_test.sh

```diff
@@ -68,6 +68,12 @@ LINKER="${LINKER:?}"
 CRYPTOLIB="${CRYPTOLIB:?}"
 RELEASE="$(lsb_release -cs)"
 
+# Ubuntu mold does not support LTO and this version also fixes a LTO+nostdlib bug.
+if [[ "$LINKER" == mold ]]; then
+    curl -L https://github.com/rui314/mold/releases/download/v2.2.0/mold-2.2.0-x86_64-linux.tar.gz | \
```

Comment:
Not the biggest find of downloading 3rd party binaries at an arbitrary version. I'd rather keep the current check till next year, when we'll get 24.04 workers with the new mold.

Reply:
It's not an arbitrary version: It's the version that fixes a bug where it throws away nostdlib entry points (no LTO involved), which will cause builds to fail. And I don't see much of a difference to the other toolchain stuff we pull.

So you'll have to either pick no mold testing at all or this; I choose the latter. I can throw in a checksum if it makes you sleep better at night.

Reply:
fine, as it will be only temporary, but please adjust the comment to specify the exact versions bugged/fixed and with a TODO to drop it once we rebase to 24.04